### PR TITLE
Add Architecture Patterns with Python book

### PR DESCRIPTION
**Pull request by threenary**
## What does this PR do?
Add resource

### Description 

Adds a book on Python

### Why is this valuable (or not)?

This resource is valuable since it introduces topics related to modern architecture patterns that many times might be overseen or left aside when trying to dive into Python.

### How do we know it's really free?

It is published by the authors on a HTML site

### For book lists, is it a book? For course lists, is it a course? etc.

This is a book.

## Checklist:
- [x] Read our [contributing guidelines](/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Review by Christianpixel-gif — CHANGES_REQUESTED** [review]
27d62d99050465c84c860e02d98f1a8bb8afe2ce

File: books/free-programming-books.md [hunk 1]
```diff
@@ -2367,6 +2367,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [A Beginner's Python Tutorial](https://en.wikibooks.org/wiki/A_Beginner%27s_Python_Tutorial) - Wikibooks
 * [A Byte of Python](https://python.swaroopch.com) (3.x) (HTML, PDF, EPUB, Mobi)
 * [A Guide to Python's Magic Methods](https://github.com/RafeKettler/magicmethods) - Rafe Kettler
+* [Architecture Patterns with Python](https://www.cosmicpython.com/book/preface.html) - Harry J.W. Percival & Bob Gregory (HTML)
```
Comment (Christianpixel-gif) [inline books/free-programming-books.md:2370]: I

**Review by Christianpixel-gif — APPROVED** [review]

**Comment by eshellman** [comment]
if our linter was working, it would complain that "architecture" should come after "A whirlwind". So to aviod breaking the build when the linter is fixed, please re-alphabetize the list.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**